### ISSUE TITLE
Several different compilation warnings

### ISSUE THREAD

**Issue by LionelCons** [fluent/fluent-bit#5171]
When compiling Fluent Bit on CentOS 7, I see several compilation warnings. Some are known problems (e.g. https://github.com/fluent/fluent-bit/issues/4427), some are new.

It is very easy to see all these warnings. Here is what I did:
```
wget https://github.com/fluent/fluent-bit/archive/refs/tags/v1.9.0.tar.gz
tar xzf v1.9.0.tar.gz
cd fluent-bit-1.9.0/build
cmake3 ..
make 2>&1 | tee build.log
grep -i -C3 warning build.log > warning.log
```

Here are some examples:
```
/tmp/fluent-bit-1.9.0/lib/cmetrics/src/cmt_decode_opentelemetry.c:114:9: warning: pointer targets in passing argument 1 of ‘cmt_variant_create_from_bytes’ differ in signedness [-Wpointer-sign]
         result_instance = cmt_variant_create_from_bytes(source->bytes_value.data, source->bytes_value.len);
         ^

/tmp/fluent-bit-1.9.0/lib/cmetrics/src/cmt_decode_opentelemetry.c:244:12: note: expected ‘char *’ but argument is of type ‘uint8_t *’
 static int append_new_metric_label_value(struct cmt_metric *metric, char *name, size_t length)
            ^

/tmp/fluent-bit-1.9.0/lib/cmetrics/src/cmt_decode_opentelemetry.c:1094:5: warning: pointer targets in passing argument 3 of ‘opentelemetry__proto__collector__metrics__v1__export_metrics_service_request__unpack’ differ in signedness [-Wpointer-sign]
     service_request = opentelemetry__proto__collector__metrics__v1__export_metrics_service_request__unpack(NULL, in_size - *offset, &in_buf[*offset]);
     ^

/tmp/fluent-bit-1.9.0/lib/monkey/mk_core/mk_event_epoll.c:382:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (index = 0 ; index < loop->n_events ; index++) {
                                ^

/tmp/fluent-bit-1.9.0/lib/monkey/mk_core/mk_event.c:109:9: warning: unused variable ‘result’ [-Wunused-variable]
     int result;
         ^
```

Since these are easy to detect, maybe this should be added as a continuous testing (for all operating systems)?

FWIW, the situation does not seem to improve. `warning.log` (as created above) contains 162 lines for 1.9.0 compared to 153 lines for 1.8.15...

**Comment by nokute78** [comment]
Note: https://github.com/fluent/fluent-bit/pull/5130 is to fix one of warnings.

**Comment by LionelCons** [comment]
FWIW, the situation does not improve. `warning.log` (as created above) now contains 220 lines for 1.9.3...

**Comment by LionelCons** [comment]
FWIW, the situation improved a tiny bit. `warning.log` (as created above) now contains 212 lines for 1.9.5...

**Comment by LionelCons** [comment]
FWIW, the situation improved a  bit. `warning.log` (as created above) now contains 153 lines for 1.9.6...

**Comment by LionelCons** [comment]
FWIW, the situation is getting slightly worse. `warning.log` (as created above) now contains 173 lines for 1.9.9...

**Comment by LionelCons** [comment]
FWIW, the situation is getting worse. `warning.log` (as created above) now contains 289 lines for 2.0.8!

**Comment by edsiper** [comment]
this has been solved while ago.

**Comment by LionelCons** [comment]
@edsiper, this problem has **not** been fixed.

I have just executed the exact same commands on the newly released 2.0.9 and I now get 265 lines in `warning.log`!

**Comment by LionelCons** [comment]
FWIW, the code quality continues to diminish.

I have just executed the exact same commands on the newly released 2.1.2 and I now get 281 lines in `warning.log`!
